flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CollectionFunctionsITCase.java

```diff
@@ -160,6 +166,117 @@ Stream<TestSetSpec> getTestSetSpecs() {
+                                "MAP_VALUES(f3)",
+                                new Map[] {Collections.singletonMap(true, "value2")},
+                                DataTypes.ARRAY(
+                                        DataTypes.MAP(DataTypes.BOOLEAN(), DataTypes.STRING()))),
+                TestSetSpec.forFunction(BuiltInFunctionDefinitions.MAP_FROM_ARRAYS, "Invalid input")
+                        .onFieldsWithData(null, null, new Integer[] {1}, new Integer[] {1, 2})
+                        .andDataTypes(
+                                DataTypes.BOOLEAN().nullable(),
+                                DataTypes.INT().nullable(),
+                                DataTypes.ARRAY(DataTypes.INT()),
+                                DataTypes.ARRAY(DataTypes.INT()))
+                        .testTableApiRuntimeError(
+                                mapFromArrays($("f2"), $("f3")),
+                                "Invalid function MAP_FROM_ARRAYS call:\n"
+                                        + "The length of the keys array 1 is not equal to the length of the values array 2")
+                        .testSqlRuntimeError(
+                                "MAP_FROM_ARRAYS(array[1, 2, 3], array[1, 2])",
+                                "Invalid function MAP_FROM_ARRAYS call:\n"
+                                        + "The length of the keys array 3 is not equal to the length of the values array 2"),
+                TestSetSpec.forFunction(BuiltInFunctionDefinitions.MAP_FROM_ARRAYS)
+                        .onFieldsWithData(
+                                new Integer[] {1, 2},
+                                new String[] {"one", "two"},
+                                new Integer[][] {new Integer[] {1, 2}, new Integer[] {3, 4}})
+                        .andDataTypes(
+                                DataTypes.ARRAY(DataTypes.INT()),
+                                DataTypes.ARRAY(DataTypes.STRING()),
+                                DataTypes.ARRAY(DataTypes.ARRAY(DataTypes.INT())))
+                        .testResult(
+                                mapFromArrays($("f0"), $("f1")),
+                                "MAP_FROM_ARRAYS(f0, f1)",
+                                of(1, "one", 2, "two"),
+                                DataTypes.MAP(DataTypes.INT(), DataTypes.STRING()))
+                        .testTableApiResult(
+                                mapFromArrays($("f1"), $("f2")),
+                                of("one", new Integer[] {1, 2}, "two", new Integer[] {3, 4}),
+                                DataTypes.MAP(
+                                        DataTypes.STRING(), DataTypes.ARRAY(DataTypes.INT()))));
+    }
+
+    // --------------------------------------------------------------------------------------------
+
+    private static <K, V> Map<K, V> of(K k1, V v1, K k2, V v2) {
+        Map<K, V> map = new HashMap<>();
+        map.put(k1, v1);
+        map.put(k2, v2);
```

Comment:
this method is not need , you can use 
`
Collections.singletonMap(
                                                Collections.singletonMap(1, 2),
                                                Collections.singletonMap(3, 4))
`
and there are some examples in MapFunctionITCase

Reply:
This one I didn't get...
One of the usages of this method is `of(1, "one", 2, "two")` to build a map with 2 entries: `1=one` and `2=two`
If I try to replace like in your suggestion then it will be `Collections.singletonMap( Collections.singletonMap(1, "one"), Collections.singletonMap(2, "two"))` and it gives a map of 2 maps each containing one entry. And this is completely different result.

Did I miss anything?

Reply:
The only thing that I currently see here is usage of `org.apache.flink.util.CollectionUtil.entry` and `org.apache.flink.util.CollectionUtil.map` which I made.
However, `Collections.singletonMap` definitely does not bring benefit here since maps of size bigger than 1 are requred here

Reply:
yeap, i just comments there are examples in MapFunctionITCase.